client.go

```diff
@@ -306,11 +316,22 @@ func (client *Client) Execute(req *Request) (*Response, error) {
 		ReceivedAt: restyResponse.ReceivedAt(),
 		Size:       restyResponse.Size(),
 		Body:       restyResponse.Body(), // byte[]
+		Error:      err,
 	}
 
 	// determine success and set err accordingly
 	if !resp.Success {
-		err = fmt.Errorf("API returned HTTP %d", resp.StatusCode)
+		// returning Request.Execute() errors
+		if resp.StatusCode == 0 {
+			var certErr x509.UnknownAuthorityError
+			if errors.As(err, &certErr) {
+				err = fmt.Errorf("%w, Remove Env Var MORPHEUS_INSECURE or set to false", err)
+			} else {
+				err = resp.Error
```

Comment:
This looks like we are returning more error information in the case where resp.StatusCode == 0
Previously we would have just returned fmt.Errorf("API returned HTTP %d", resp.StatusCode)
Are there any cases where resp.Error may contain sensitive information? If we are not sure it may
be best to preserve the existing behaviour and potentially look at that change in a seperate pr.